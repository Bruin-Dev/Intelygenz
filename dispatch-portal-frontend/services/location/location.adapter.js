export const locationInAdapter = data => ({
  owner: data.client_name,
  address1: data.client_address.address,
  country: data.client_address.country,
  city: data.client_address.city,
  state: data.client_address.state,
  zip: data.client_address.zip
});
