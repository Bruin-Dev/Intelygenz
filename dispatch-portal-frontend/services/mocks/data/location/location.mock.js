export const getLocationByTicketIdMock = {
  client_name: 'Red Rose',
  client_address: {
    address: '123 Fake Street',
    city: 'Pleasantown',
    state: 'New York',
    zip: '99085',
    country: 'United States'
  }
};
