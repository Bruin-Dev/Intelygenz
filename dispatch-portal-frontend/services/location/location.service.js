import { API_URLS } from '../api.config';
import axiosInstance from '../api';
import { locationInAdapter } from './location.adapter';

export class LocationService {
  constructor(axiosAuxI = axiosInstance) {
    this.axiosI = axiosAuxI;
  }

  async getLocationByTicketId(ticketId) {
    try {
      const res = await this.axiosI.get(
        `${API_URLS.GET_LOCATION_BY_TICKET_ID}/${ticketId}`
      );
      return locationInAdapter(res.data);
    } catch (error) {
      return this.captureErrorGeneric(error);
    }
  }

  // eslint-disable-next-line class-methods-use-this
  captureErrorGeneric(error) {
    return { error: error.message };
  }
}
