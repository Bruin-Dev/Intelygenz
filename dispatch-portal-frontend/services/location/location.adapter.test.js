import { locationInAdapter } from './location.adapter';
import { getLocationByTicketIdMock } from '../mocks/data/location/location.mock';

describe('location adapter tests', () => {
  it('should match the snapshot', () => {
    expect(locationInAdapter(getLocationByTicketIdMock)).toMatchSnapshot();
  });
});
