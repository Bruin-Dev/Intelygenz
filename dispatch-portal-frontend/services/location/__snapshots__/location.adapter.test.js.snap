// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`location adapter tests should match the snapshot 1`] = `
Object {
  "address1": "123 Fake Street",
  "city": "Pleasantown",
  "country": "United States",
  "owner": "Red Rose",
  "state": "New York",
  "zip": "99085",
}
`;
